runtime/client_http_request_test.go

```diff
@@ -321,6 +321,39 @@ func TestMakingClientCallWithThriftException(t *testing.T) {
 	assert.Len(t, logs["Finished an outgoing client HTTP request"], 1)
 }
 
+func TestMakingListAndEnumClientCallWithSuccess(t *testing.T) {
+	gateway, err := benchGateway.CreateGateway(
+		defaultTestConfig,
+		defaultTestOptions,
+		exampleGateway.CreateGateway,
+	)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer gateway.Close()
+
+	bgateway := gateway.(*benchGateway.BenchGateway)
+
+	bgateway.HTTPBackends()["bar"].HandleFunc(
+		"GET", "/bar/list-and-enum",
+		func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			_, _ = w.Write([]byte(`"test-response"`))
+		},
+	)
+
+	deps := bgateway.Dependencies.(*exampleGateway.DependenciesTree)
+	bClient := deps.Client.Bar
+
+	body, _, err := bClient.ListAndEnum(
```

Comment:
This test does not make sense:
1. the thrift change covers both endpoint and client, a request should be made to the endpoint not the client
2. there is no http query in this test, the code that does `http query -> enum type` is not tested at all
3. there is no change with http client runtime, it doesn't make sense to add tests here. The right place would be `thrift_test.go` and `examples/example-gateway/endpoints/bar`
4. a good example for writing better and easier integration tests is https://github.com/uber/zanzibar/blob/master/examples/example-gateway/endpoints/contacts/save_contacts_test.go#L15

Reply:
this test is needed bcs we want to test client is sending the right params to the client backend. have added the params though in the request and checking in client backend. Also, added a endpoint test case which mocks client which is insufficient in testing whether client gen code works correctly. thus, two test cases are needed. Or an integration test which can start a client backend and then test e2e would also be good.